app/services/payments/customer.rb

```diff
@@ -45,11 +45,14 @@ def charge(customer:, amount:, description:, card_id: nil)
       def request
         yield
       rescue Stripe::InvalidRequestError => e
+        DataDogStatsClient.increment("stripe.errors")
```

Comment:
I think this means that the error is logged twice, as you also log it here https://github.com/thepracticaldev/dev.to/pull/6188/files#diff-a6c4de2161780834712084552a4fb927R19

maybe both calls to DataDog should reside here :)

Reply:
I threw in a quick `puts` here and then ran the spec for `/app/controllers/stripe_active_cards_controller.rb` and I didn't see them triggered. Where is the reference that makes this get logged twice?

Reply:
My suggestion was because I read the code like this:

- Payments::Customer gets called, raises an exception, then enters here:

https://github.com/thepracticaldev/dev.to/pull/6188/files#diff-8dbfc7bc2cdbed97f7900a67b400e85aR51

which logs to DD, then raises the exception `CardError`, which, inside the controller gets rescued and then logs again to DD:

https://github.com/thepracticaldev/dev.to/pull/6188/files#diff-a6c4de2161780834712084552a4fb927R39-R41

That's why I suggested to keep all the DD logging inside the `Payments::Customer` object's `rescue`, unless you need to send data that's available only in the controller

Let me know if I understood correctly :D

Reply:
I think it's me not understanding correctly. There are 2 points I'm unclear on:
1) Where does `Payments::Customer` get invoked from the `stripe_active_cards_controller` which would cause the error twice? I see`Payments::Customer` called in `credits_controller`, but I didn't add anything for DataDog in that controller.
2) We'll still need it in the `stripe_active_cards_controller` for `stripe.errors.update_subscription` and `stripe.errors.new_subscription`. Or is it safe to associate a `CardError` to `update_subscription` and `InvalidRequestError` to `new_subscription`?

Reply:
I understand the issue of communication, sorry I wasn't clear. I think you have to rebase your PR to understand what I mean, the current `StripeActiveCardsController` merged with #6181 **only** uses `Payments::Customer` in all controller, so if you move `DataDogStatsClient.increment("stripe.errors")` from all exception handlers in the controller in there, you'll have it for all Stripe calls, neat in one place.

Reply:
Ahhhhhhhh I see, I wasn't looking at #6181. Makes sense now, thanks for the clarification!